libguard/guard_common.hpp

```diff
@@ -1,17 +1,21 @@
 // SPDX-License-Identifier: Apache-2.0
 #pragma once
 
+#include "guard_exception.hpp"
 #include "guard_log.hpp"
 
 #include <array>
 #include <cstdint>
-#include <stdexcept>
 #include <optional>
+#include <stdexcept>
```

Comment:
General comment: all the exported api's ( in guard_interface.hpp) should updated with possible exception list. instead of making generic statement.  "  Throw exception on failure".

Reply:
As a practice applying this to all the functions throwing custom exceptions.